ddl/index.go

```diff
@@ -1119,72 +1111,84 @@ func (w *worker) addPhysicalTableIndex(t table.PhysicalTable, indexInfo *model.I
 	}
 
 	// variable.ddlReorgWorkerCounter can be modified by system variable "tidb_ddl_reorg_worker_cnt".
-	workerCnt := variable.GetDDLReorgWorkerCounter()
+	workerCnt := int(variable.GetDDLReorgWorkerCounter())
 	idxWorkers := make([]*addIndexWorker, 0, workerCnt)
 	defer func() {
 		closeAddIndexWorkers(idxWorkers)
 	}()
+	availableWorkerCh := make(chan *addIndexWorker, variable.MaxDDLReorgWorkerCount) // max worker count
+	doingTaskCh := make(chan *reorgIndexTask, variable.MaxDDLReorgWorkerCount)       // max worker count
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
```

Comment:
assign this goroutine function to a local variable, and start it, like:
```
dispatchTaskFunc := {
	....
}

go dispatchTaskFunc()
```

Make the code be clearer.